File: Demos/ShapePainter/DrawingBoard.cs
using System;
using System.Collections;
using System.Drawing;
using System.Windows.Forms;

namespace ShapePainter
{
	/// <summary>
	/// A panel that draws shapes.
	/// </summary>
	public class DrawingBoard: Panel
	{
		Shapes m_Shapes;

		public Shapes Shapes
		{
			get
			{
				return m_Shapes;
			}
			set
			{
				m_Shapes = value;
			}
		}

		protected override void OnPaint(PaintEventArgs e)
		{
			if (this.m_Shapes != null)
			{
				Graphics g = e.Graphics;
				foreach(Shape shape in this.m_Shapes)
				{
					shape.draw(g);
				}
			}
		}
	}
}

File: Demos/ShapePainterVB/DrawingBoard.vb
Imports System
Imports System.Collections
Imports System.Drawing
Imports System.Windows.Forms
Namespace ShapePainter

	Public Class DrawingBoard
	Inherits Panel
		Private m_Shapes As Shapes

		Public Property Shapes() As Shapes
			Get
				Return m_Shapes
			End Get
			Set
				m_Shapes = value
			End Set
		End Property

		Protected Overloads Overrides Sub OnPaint(ByVal e As PaintEventArgs)
			If Not (Me.m_Shapes Is Nothing) Then
                Dim g As Graphics = e.Graphics
                Dim MyShape As Shape
                For Each MyShape In Me.m_Shapes
                    MyShape.draw(g)
                Next
            End If
		End Sub
	End Class 
End Namespace
